src/guiapi/src/MItem_menus.cpp

```diff
@@ -218,3 +220,38 @@ MI_FOOTER_SETTINGS::MI_FOOTER_SETTINGS()
 void MI_FOOTER_SETTINGS::click(IWindowMenu & /*window_menu*/) {
     Screens::Access()->Open(GetScreenMenuFooterSettings);
 }
+
+MI_LANGUAGUE_USB::MI_LANGUAGUE_USB()
+    : WI_LABEL_t(_(label), 0, is_enabled_t::yes, is_hidden_t::no) {}
+
+void MI_LANGUAGUE_USB::click(IWindowMenu &windowMenu) {
+    if (fileProviderUSB.OpenFile())
+        ProviderRegistrator("ts", &fileProviderUSB);
+}
+
+MI_LOAD_LANG::MI_LOAD_LANG()
+    : WI_LABEL_t(_(label), 0, is_enabled_t::yes, is_hidden_t::no) {}
+
+void MI_LOAD_LANG::click(IWindowMenu &windowMenu) {
+
+    uint8_t buff[512];
+
+    FILE *srcDir = fopen("/usb/lang/ts.mo", "rb");
+    FILE *dstDir = fopen("/internal/ts.mo", "wb");
+    //copy languague from usb to xflash
+    if (dstDir && srcDir) {
+        for (size_t readBytes = fread(buff, 1, 16, srcDir); readBytes != 0; readBytes = fread(buff, 1, 16, srcDir)) {
+            fwrite(buff, 1, readBytes, dstDir);
+            fflush(dstDir);
```

Comment:
is the flush still necessary?

Reply:
Yes it is neccesary. Beacuse littlefs will write to the hardware medium only when flush or close is called and we have 16B buffer, thus we need to flush between each write.

Reply:
ok, so this remains to be fixed in littlefs

Reply:
After further testing it seems to weok without flush